server/src/services/library.service.ts

```diff
@@ -366,258 +358,181 @@ export class LibraryService {
     return JobStatus.SUCCESS;
   }
 
-  async handleAssetRefresh(job: ILibraryFileJob): Promise<JobStatus> {
+  async handleSyncFile(job: ILibraryFileJob): Promise<JobStatus> {
+    // Only needs to handle new assets
     const assetPath = path.normalize(job.assetPath);
 
-    const existingAssetEntity = await this.assetRepository.getByLibraryIdAndOriginalPath(job.id, assetPath);
+    let asset = await this.assetRepository.getByLibraryIdAndOriginalPath(job.id, assetPath);
+    if (asset) {
+      return JobStatus.SKIPPED;
+    }
 
-    let stats: Stats;
+    let stat;
     try {
-      stats = await this.storageRepository.stat(assetPath);
-    } catch (error: Error | any) {
-      // Can't access file, probably offline
-      if (existingAssetEntity) {
-        // Mark asset as offline
-        this.logger.debug(`Marking asset as offline: ${assetPath}`);
-
-        await this.assetRepository.update({ id: existingAssetEntity.id, isOffline: true });
-        return JobStatus.SUCCESS;
-      } else {
-        // File can't be accessed and does not already exist in db
-        throw new BadRequestException('Cannot access file', { cause: error });
+      stat = await this.storageRepository.stat(assetPath);
+    } catch (error: any) {
+      if (error.code === 'ENOENT') {
+        this.logger.error(`File not found: ${assetPath}`);
+        return JobStatus.SKIPPED;
       }
+      this.logger.error(`Error reading file: ${assetPath}. Error: ${error}`);
+      return JobStatus.FAILED;
     }
 
-    let doImport = false;
-    let doRefresh = false;
+    this.logger.log(`Importing new library asset: ${assetPath}`);
 
-    if (job.force) {
-      doRefresh = true;
+    const library = await this.repository.get(job.id, true);
+    if (library?.deletedAt) {
```

Comment:
Shouldn't we check if the library actually exists? I guess it's already covered wherever we queue the sync, but it still feels odd not sanity-checking that here